private/buf/cmd/buf/command/push/push.go

```diff
@@ -154,7 +154,8 @@ func (f *flags) Bind(flagSet *pflag.FlagSet) {
 
 The source control URL and default branch is based on the required Git remote %q.
 This flag is only compatible with checkouts of Git source repositories.
-This flag does not allow you to set any of the following flags yourself: --%s, --%s.`,
+If you set the --%s flag yourself, then this value will be used instead and the source control URL will not be derived from the Git source control state.
```

Comment:
Are there other flags we want to be allowed to be overwritten?

Reply:
So the only flag left is `--create-default-label`. So a use-case hasn't come up for that flag yet (we had a use-case come up for `--source-control-url` for testing purposes), and I am comfortable not loosening this pre-emptively, since the general rule is that it's much easier to loosen something than to make something more restrictive down the road. However that isn't a strong opinion for me, if others feel that we should simply allow overrides for both flags (and we already have `--label` be additive to the labels `--git-metadata` is setting) as a simpler mental model, I would be onboard with this.

Reply:
I would provide overrides consistently here, so I would do it too

Reply:
Sounds good, I'll adjust!